TLM/TLM/Util/Extensions/NetSegmentExtensions.cs

```diff
@@ -9,5 +53,130 @@ public static class NetSegmentExtensions {
+            var laneList = new List<LanePos>();
+
+            bool inverted = (netSegment.m_flags & NetSegment.Flags.Invert) != NetSegment.Flags.None;
+
+            NetInfo.Direction? filterDir = null;
+            NetInfo.Direction sortDir = NetInfo.Direction.Forward;
+
+            if (startNode != null) {
+                filterDir = (bool)startNode
+                                ? NetInfo.Direction.Backward
+                                : NetInfo.Direction.Forward;
+                filterDir = inverted
+                                ? NetInfo.InvertDirection((NetInfo.Direction)filterDir)
+                                : filterDir;
+                sortDir = NetInfo.InvertDirection((NetInfo.Direction)filterDir);
+            } else if (inverted) {
+                sortDir = NetInfo.Direction.Backward;
+            }
+
+            if (reverse) {
+                sortDir = NetInfo.InvertDirection(sortDir);
+            }
+
+            NetInfo segmentInfo = netSegment.Info;
+            uint curLaneId = netSegment.m_lanes;
+            byte laneIndex = 0;
+
+            while (laneIndex < segmentInfo.m_lanes.Length && curLaneId != 0u) {
+                NetInfo.Lane laneInfo = segmentInfo.m_lanes[laneIndex];
+                if ((laneTypeFilter == null ||
+                     (laneInfo.m_laneType & laneTypeFilter) != NetInfo.LaneType.None) &&
+                    (vehicleTypeFilter == null || (laneInfo.m_vehicleType & vehicleTypeFilter) !=
+                     VehicleInfo.VehicleType.None) &&
+                    (filterDir == null ||
+                     segmentInfo.m_lanes[laneIndex].m_finalDirection == filterDir)) {
+                    laneList.Add(
+                        new LanePos(
+                            curLaneId,
+                            laneIndex,
+                            segmentInfo.m_lanes[laneIndex].m_position,
+                            laneInfo.m_vehicleType,
+                            laneInfo.m_laneType));
+                }
+
+                curLaneId = netManager.m_lanes.m_buffer[curLaneId].m_nextLane;
+                ++laneIndex;
+            }
+
+            if (sort) {
+                int CompareLanePositionsFun(LanePos x, LanePos y) {
+                    bool fwd = sortDir == NetInfo.Direction.Forward;
+                    if (Math.Abs(x.position - y.position) < 1e-12) {
+                        if (x.position > 0) {
+                            // mirror type-bound lanes (e.g. for coherent disply of lane-wise speed limits)
+                            fwd = !fwd;
+                        }
+
+                        if (x.laneType == y.laneType) {
+                            if (x.vehicleType == y.vehicleType) {
+                                return 0;
+                            }
+
+                            if ((x.vehicleType < y.vehicleType) == fwd) {
+                                return -1;
+                            }
+
+                            return 1;
```

Comment:
ternary op?

Reply:
why do you like ternary op so much anyway? I find it confusing most of the time. here I prefer not to change the code since it can introduce mistake.

Reply:
I just find it to be much more concise and, when separated on to multiple lines, much easier to read, eg:

```csharp
return ((x.vehicleType < y.vehicleType) == fwd)
    ? -1
    : 1;
```

vs.

```csharp
if ((x.vehicleType < y.vehicleType) == fwd) {
    return -1;
}

return 1;
```

Reply:
in general I don't want to touch anything in `GetSortedLanes` in a PR like this because the function is big and complex and has too many parantheses therefore it introduces chance of failure.